### DISCUSSION TITLE
How to best do fuzzy comparison for floats

### DISCUSSION THREAD

**Discussion by define-private-public** [halide/Halide#5826] [Q&A] [answered]
I was porting an algorithm from C++ over to use Halide.  At some point in the algorithm, there was this

```
float f = //...;
//...
if (f == 0.0) {
//...
```

I know that is can be a little dangerous to compare floats to exact numbers, which is why you should do fuzzy comparison.  Is there a recommended way to do this in Halide?

**Comment by abadams** [comment, marked as answer]
There's no Halide-specific way to handle this. The idioms are similar to C/C++. Depending on what kind of error I'm being robust to, I usually do something like `abs(f) < 0.001f`
